src/libutil/executable-path.cc

```diff
@@ -0,0 +1,67 @@
+#include "environment-variables.hh"
+#include "executable-path.hh"
+#include "strings-inline.hh"
+#include "util.hh"
+#include "file-path-impl.hh"
+
+namespace nix {
+
+namespace fs = std::filesystem;
+
+constexpr static const OsStringView pATH_separator =
+#ifdef WIN32
+    L";"
+#else
+    ":"
+#endif
+    ;
+
+std::vector<fs::path> parsePATH()
+{
+    return parsePATH(getEnvNative(PATHNG_LITERAL("PATH")).value_or(PATHNG_LITERAL("")));
+}
+
+std::vector<fs::path> parsePATH(const OsString & path)
+{
+    auto strings = basicTokenizeString<std::list<OsString>, OsString::value_type>(path, pATH_separator);
```

Comment:
This is untested and broken.
Feel free to borrow everything from https://github.com/NixOS/nix/pull/11021/commits/a4b489dd1a7fd579e84670291f0a6a318e12a9b9 (part of #11021). That was a rabbit hole (https://github.com/NixOS/nix/pull/11093) I haven't finished climbing back out of, because of other priorities, but I see that you need it here too.

Reply:
I punted on the path lookup case (with an assert) (as opposed to name look up case) since I am not sure sort of error message we want to do, and as it turns out in most cases we are just looking up statically known names.

This brings us back into compliance. I think just in the next PR, for the build hook, we look up a dynamic choice. So we can see what we want to do then.